syntax = "proto3";



message Book {
    string author = 1;
    string title = 2;
    uint64 isbn = 3;
}


message GetBooksRequest{
    
}


message GetBooksResponse{

}

// message GetReturnDeadlineRequest{
//     string uuid = 1;
// }


// message GetReturnDeadlineResponse{
//     oneof date_or_err{
//         string date = 1;
//         Error err = 2;
//     }
// }


// message GetBookCountRequest{
//     string uuid = 1; 
// }


// message GetBookCountResponse{
//     oneof book_count_or_err{
//         uint32 book_count = 1;
//         Error err = 2;
//     }
// }


message Error {
  string message = 1;
  uint32 code = 2;
}

service Library {
  rpc GetBooks (GetBooksRequest) returns (GetBooksResponse);
  rpc GetReturnDeadline (GetReturnDeadlineRequest) returns (GetReturnDeadlineResponse);
  rpc GetBookCount (GetBookCountRequest) returns (GetBookCountResponse);
}

